UnityAds/src/main/java/com/mopub/mobileads/UnityBaseAd.java

```diff
@@ -94,21 +67,18 @@ protected void load(@NonNull Context context, @NonNull AdData adData) {
         Preconditions.checkNotNull(context);
         Preconditions.checkNotNull(adData);
 
-        if (!UnityAds.isInitialized()) {
-            initializeAndFailAdRequest(context, adData);
-            return;
-        }
-
         final Map<String, String> extras = adData.getExtras();
 
+        UnityAdsInitializer.getInstance().checkOrInitUnityAds((Activity) context, extras);
+
         mPlacementId = placementIdForServerExtras(extras);
         setAutomaticImpressionAndClickTracking(false);
     }
 
     // Returns true if placementId is null and reports the load failure
     public boolean failIfPlacementIdNull() {
         if (mPlacementId == null) {
-            MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_PLACEMENT_ID_MISSING);
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, LOG_PLACEMENT_ID_MISSING.getMessage());
```

Comment:
since the Enum toString() is overridden already, is this redundant?

Reply:
I think we need getMessage here otherwise the enum number (0, 1, 2 etc) will get sent